### DISCUSSION TITLE
Build Plugin: Adding Custom Additional Types

### DISCUSSION THREAD

**Discussion by dhartung** [ExpediaGroup/graphql-kotlin#1509] [Q&A]
Hi there! I read in the documentation that you can register [additional types](https://opensource.expediagroup.com/graphql-kotlin/docs/schema-generator/customizing-schemas/advanced-features) which do not appear in a query, mutation or subscription. This is awesome as I need this feature for types that are only used in resolveres for Federation.

However, I can't find the correct hook to access or override the SchemaGenerator like described in the documentation when using the Gradle Build Plugin. I was able to provide a SchemaGeneratorHooksProvider to define custom scalars, but I need a pointer where to look to access the actual SchemaGenerator :)

Thanks in advance!

**Comment by dariuszkuc** [comment]
Hello 👋 
Federation is already supported (see [docs](https://opensource.expediagroup.com/graphql-kotlin/docs/schema-generator/federation/apollo-federation) - any types annotated with `@ExtendsDirective` will be automatically picked up and processed as if they were exposed as if they were accessible through regular query/mutation path.


You could try overriding the `fun willBuildSchema(builder: GraphQLSchema.Builder): GraphQLSchema.Builder` hook and manually add the types in there but it won't let you to call in to the schema generator methods.
